### DISCUSSION TITLE
Get MilvusException (unrecoverable error) when search, What are the possible reasons?

### DISCUSSION THREAD

**Discussion by simonwei97** [milvus-io/milvus#27845] [Q&A and General discussion]
## Milvus Configure

- version: `v2.2.13`
- deploymode: standalone

## What happen

Before searching, I run benchmark on process service (it responsible for for resolving file and embeding file text chunks into Milvus), which operation **almost drained the machine's CPU**. It lead some services (including Milvus, DB, etc) hang on, which seems like stuck. Then I restarted all services for searching, but I found that milvus restarted several times.

![image](https://github.com/milvus-io/milvus/assets/119845914/1582ddfb-20f6-449d-8982-bb94b57537b6)

![image](https://github.com/milvus-io/milvus/assets/119845914/303f2436-e096-4284-a9a9-e577d1754036)

## Search

I picked different collections for searching, some collections searching raised the following errors, others results was ok.

- search client error
```bash
Error occurred!
<MilvusException: (code=1, message=attempt #0: fail to get shard leaders from QueryCoord: collection=444343578607529465: collection not loaded: unrecoverable error: fail to query on all shard leaders)>
```

- Milvus server error
![image](https://github.com/milvus-io/milvus/assets/119845914/a47ad871-1518-42fd-ae9e-9afd441f8285)

## Question
1. Benchmark operation lead some collections going to unavailable status. Why ?
2. How can I do for solving this sticky situation ?
3. How to avoid this situation in production environment ?

**Comment by yhmo** [comment]
The datacoord failed to initialize. The error "fail to init rocksmq [Error] While lock file /var/lib/milvus/rdb_data_meta_kv/LOCK: Resource temporarily unavailable"

The "Resource temporarily unavailable" error is returned from the Linux OS. 
Possible reason: 
- Some applications open a large number of processes. The number of processes is reaching the kernel max value.
- Some applications open too many files.
- The server is running out of memory. 

I guess your program opened too many files.

**Reply by simonwei97** [reply]
I restarted servies (including milvus), but this error still occured.

**Comment by simonwei97** [comment]
more milvus log for searching when client got failed error
<img width="1791" alt="image" src="https://github.com/milvus-io/milvus/assets/119845914/ce5fddaa-1a2c-49f0-9fb8-581f4de73fc7">

**Reply by yhmo** [reply]
"127.0.0.1:2379: error: connection refused" ----------------------------- etcd service is down or cannot access
